packages/adblocker-electron-preload/src/index.ts

```diff
@@ -7,24 +7,20 @@
  */
 
 import { ipcRenderer } from 'electron';
+import { DOMMonitor, IBackgroundCallback } from '@cliqz/adblocker-content';
 
-import {
-  DOMMonitor,
-  IBackgroundCallback,
-  IMessageFromBackground,
-  injectScript,
-} from '@cliqz/adblocker-content';
-
-function getCosmeticsFiltersFirst(): string[] | null {
-  return ipcRenderer.sendSync('get-cosmetic-filters-first', window.location.href);
+interface CosmeticFiltersResponse {
+  active: boolean;
+  error?: string;
 }
-function getCosmeticsFiltersUpdate(data: Omit<IBackgroundCallback, 'lifecycle'>) {
-  ipcRenderer.send('get-cosmetic-filters', window.location.href, data);
+
+function injectCosmeticFilters(data?: Omit<IBackgroundCallback, 'lifecycle'>): Promise<CosmeticFiltersResponse> {
+  return ipcRenderer.invoke('inject-cosmetic-filters', window.location.href, data);
 }
 
-if (window === window.top && window.location.href.startsWith('devtools://') === false) {
-  (() => {
-    const enableMutationObserver = ipcRenderer.sendSync('is-mutation-observer-enabled');
+if (window === window.top && !window.location.href.startsWith('devtools://')) {
```

Comment:
This change is not required, please revert this back to `window === window.top && window.location.href.startsWith('devtools://') === false`. I don't feel the problem but this is just our internal preference to make things more explicit. 👍